docs/examples/_server.py

```diff
@@ -169,19 +172,48 @@
 class MockData(DataServiceBase):
     def __init__(self):
         self.tabular_data_requested = False
-        self.tabular_response = [{"PowerPct": 0, "IsPowered": False}, {"PowerPct": 0, "IsPowered": False}, {"Position": 0}]
+        self.tabular_response = [
+            DataClient.TabularData(
+                {"PowerPct": 0, "IsPowered": False},
+                CaptureMetadata(method_name="IsPowered"),
+                datetime(2022, 1, 1, 1, 1, 1),
+                datetime(2022, 12, 31, 23, 59, 59),
+            ),
+            DataClient.TabularData(
+                {"PowerPct": 0, "IsPowered": False},
+                CaptureMetadata(location_id="loc-id"),
+                datetime(2023, 1, 2),
+                datetime(2023, 3, 4)
+            ),
+            DataClient.TabularData(
+                {"Position": 0},
+                CaptureMetadata(),
+                datetime(2023, 5, 6),
+                datetime(2023, 7, 8),
+            ),
+        ]
 
     async def TabularDataByFilter(self, stream: Stream[TabularDataByFilterRequest, TabularDataByFilterResponse]) -> None:
         if self.tabular_data_requested:
             await stream.send_message(TabularDataByFilterResponse())
             return
         self.tabular_data_requested = True
         _ = await stream.recv_message()
-        n = len(self.tabular_response)
-        tabular_structs = [Struct()] * n
-        for i in range(n):
-            tabular_structs[i].update(self.tabular_response[i])
-        await stream.send_message(TabularDataByFilterResponse(data=[TabularData(data=struct) for struct in tabular_structs]))
+        tabular_structs = []
+        tabular_metadata = [data.metadata for data in self.tabular_response]
+        for idx, tabular_data in enumerate(self.tabular_response):
+            tabular_structs.append(
+                TabularData(
+                    data=dict_to_struct(tabular_data.data),
+                    metadata_index=idx,
+                    time_requested=datetime_to_timestamp(tabular_data.time_requested),
+                    time_received=datetime_to_timestamp(tabular_data.time_received)
+                )
+            )
+        await stream.send_message(TabularDataByFilterResponse(
```

Comment:
(q) Where is this message being sent?

Reply:
Good question! The message is being sent back along the channel where the request was sent, so basically back to the calling client.